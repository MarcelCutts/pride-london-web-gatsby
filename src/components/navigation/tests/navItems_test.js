import React from 'react'
import { shallow } from 'enzyme'
import NavItems from '../navItems'
import shortid from 'shortid' //eslint-disable-line no-unused-vars
jest.mock('shortid', () => ({ generate: jest.fn().mockReturnValue('mockId') }))

describe('header', () => {
  it('matches snapshot', () => {
    const mockProps = {
      items: {
        listItems: ['Shante, you stay', 'Sashay away'],
      },
    }
    const wrappedComponent = shallow(<NavItems {...mockProps} />)
    expect(wrappedComponent).toMatchSnapshot()
  })
  it('presents three links when three listItems passed', () => {
    const mockProps = {
      items: {
        listItems: [
          'Shante, you stay',
          'Sashay away',
          "how can you love someone else if you don't love yourself",
        ],
      },
    }
    const wrappedComponent = shallow(<NavItems {...mockProps} />)
    expect(wrappedComponent.find('LinkTest')).toHaveLength(3)
  })
  it('presents one button', () => {
    const mockProps = {
      items: {
        listItems: [],
      },
    }
    const wrappedComponent = shallow(<NavItems {...mockProps} />)
    expect(wrappedComponent.find('ButtonTest')).toHaveLength(1)
  })
})
