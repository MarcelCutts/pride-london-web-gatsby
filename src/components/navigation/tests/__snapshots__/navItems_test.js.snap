// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavItems
    items={
      Object {
        "listItems": Array [
          "Shante, you stay",
          "Sashay away",
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Column
          alignItems="center"
          flex={3}
          justifyContent="center"
        >
          <img
            alt=""
            src={undefined}
          />
        </Column>,
        Array [
          <Column>
            <LinkTest
              to="#"
            >
               
              Shante, you stay
            </LinkTest>
          </Column>,
          <Column>
            <LinkTest
              to="#"
            >
               
              Sashay away
            </LinkTest>
          </Column>,
        ],
        <Column>
          <ButtonTest
            to="/"
          >
            Donate
          </ButtonTest>
        </Column>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "alignItems": "center",
          "children": <img
            alt=""
            src={undefined}
          />,
          "flex": 3,
          "justifyContent": "center",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "H1jvaCLkX",
        "nodeType": "function",
        "props": Object {
          "children": <LinkTest
            to="#"
          >
             
            Shante, you stay
          </LinkTest>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              " ",
              "Shante, you stay",
            ],
            "to": "#",
          },
          "ref": null,
          "rendered": Array [
            " ",
            "Shante, you stay",
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "BJgowTC8k7",
        "nodeType": "function",
        "props": Object {
          "children": <LinkTest
            to="#"
          >
             
            Sashay away
          </LinkTest>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              " ",
              "Sashay away",
            ],
            "to": "#",
          },
          "ref": null,
          "rendered": Array [
            " ",
            "Sashay away",
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ButtonTest
            to="/"
          >
            Donate
          </ButtonTest>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Donate",
            "to": "/",
          },
          "ref": null,
          "rendered": "Donate",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Column
            alignItems="center"
            flex={3}
            justifyContent="center"
          >
            <img
              alt=""
              src={undefined}
            />
          </Column>,
          Array [
            <Column>
              <LinkTest
                to="#"
              >
                 
                Shante, you stay
              </LinkTest>
            </Column>,
            <Column>
              <LinkTest
                to="#"
              >
                 
                Sashay away
              </LinkTest>
            </Column>,
          ],
          <Column>
            <ButtonTest
              to="/"
            >
              Donate
            </ButtonTest>
          </Column>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "alignItems": "center",
            "children": <img
              alt=""
              src={undefined}
            />,
            "flex": 3,
            "justifyContent": "center",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "",
              "src": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "H1jvaCLkX",
          "nodeType": "function",
          "props": Object {
            "children": <LinkTest
              to="#"
            >
               
              Shante, you stay
            </LinkTest>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                " ",
                "Shante, you stay",
              ],
              "to": "#",
            },
            "ref": null,
            "rendered": Array [
              " ",
              "Shante, you stay",
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "BJgowTC8k7",
          "nodeType": "function",
          "props": Object {
            "children": <LinkTest
              to="#"
            >
               
              Sashay away
            </LinkTest>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                " ",
                "Sashay away",
              ],
              "to": "#",
            },
            "ref": null,
            "rendered": Array [
              " ",
              "Sashay away",
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <ButtonTest
              to="/"
            >
              Donate
            </ButtonTest>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Donate",
              "to": "/",
            },
            "ref": null,
            "rendered": "Donate",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
